### PR TITLE
[jit] add reverse to list

### PR DESCRIPTION
**Pull request by TheCodez**
Add reverse functionality to list. See https://github.com/pytorch/pytorch/issues/16662

```python
import torch

@torch.jit.script
def foo():
    a = [1, 2, 3, 4]
	a.reverse()

    return a
```

### CODE REVIEW

**Review by driazati — APPROVED** [review]
Looks good! Pinging @eellison for the alias annotations

File: torch/csrc/jit/register_prim_ops.cpp [hunk 1]
```diff
@@ -1003,6 +1004,20 @@ Operation listAppend(const Node* node) {
   };
 }
 
+template <typename TList>
+Operation listReverse(const Node* node) {
+  return [](Stack& stack) {
```
Comment (TheCodez) [inline torch/csrc/jit/register_prim_ops.cpp]: @driazati should I use `std::function<int(Stack&)>` directly here as well?

**Pull request closed without merging**